execution/block-partitioner/src/pre_partition/mod.rs

```diff
@@ -0,0 +1,25 @@
+// Copyright © Aptos Foundation
+
+use crate::v2::types::PrePartitionedTxnIdx;
+use aptos_types::transaction::analyzed_transaction::AnalyzedTransaction;
+
+pub trait PrePartitioner: Send {
+    /// The initial partitioning phase for `ShardedBlockPartitioner`/`PartitionerV2` to divide a block into `num_shards` sub-blocks.
+    /// See `PartitionerV2::partition()` for more details.
+    fn pre_partition(
+        &self,
+        transactions: &[AnalyzedTransaction],
+        num_shards: usize,
+    ) -> Vec<Vec<PrePartitionedTxnIdx>>;
+}
+
+pub mod uniform_partitioner;
+
+pub fn start_txn_idxs(pre_partitioned: &Vec<Vec<PrePartitionedTxnIdx>>) -> Vec<PrePartitionedTxnIdx> {
```

Comment:
Looks correct, but probably good idea to have a unit test for this function.

Reply:
fixed